### PR TITLE
feature: COIOS-413 Add configuration screen to the demo UIKit app

### PR DESCRIPTION
**Pull request by kaphacius**
This PR will add a configuration screen to the Demo App.
Things missing:
- search for country/currency selection
- index list for country/currency selection
- ability to reset the configuration if current one is causing an error alert loop
- better naming :)

### CODE REVIEW

File: Adyen/Formatters/AmountFormatter.swift [hunk 1]
```diff
@@ -54,7 +54,12 @@ public final class AmountFormatter {
             .rounding(accordingToBehavior: roundTowardsZero).intValue
     }
     
-    private static func decimalAmount(_ amount: Int, currencyCode: String) -> NSDecimalNumber {
+    /// Converts an amount in major currency unit to a decimal amount in major currency units.
```
Comment (mohammedDehairy) [inline Adyen/Formatters/AmountFormatter.swift:57]: minor to major, not major to major :-)

**Review by mohammedDehairy — APPROVED** [review]

File: Adyen/Formatters/AmountFormatter.swift (hunk 1 above)
Comment (kaphacius) [reply Adyen/Formatters/AmountFormatter.swift:57]: 👍

**Review by descorp — DISMISSED** [review]

File: Adyen/Formatters/AmountFormatter.swift [hunk 2]
```diff
@@ -54,7 +54,12 @@ public final class AmountFormatter {
             .rounding(accordingToBehavior: roundTowardsZero).intValue
```
Comment (descorp) [inline Adyen/Formatters/AmountFormatter.swift:54]: BTW, `AmountFormatter ` coulld be enum

**Review by mohammedDehairy — APPROVED** [review]

File: Demo/Common/Configuration/SearchBar.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        @Binding private var text: String
+
+        fileprivate init(text: Binding<String>) {
+            _text = text
+        }
+
+        internal func searchBar(
+            _ searchBar: UISearchBar,
+            textDidChange searchText: String
+        ) {
```
Comment (descorp) [inline Demo/Common/Configuration/SearchBar.swift:51]: why new line ?

Comment (kaphacius) [reply Demo/Common/Configuration/SearchBar.swift:51]: One line was getting too long, did not look nice :)

**Review by descorp — APPROVED** [review]

File: Demo/Common/Configuration/SearchBar.swift (hunk 3 above)
Comment (descorp) [reply Demo/Common/Configuration/SearchBar.swift:51]: Than why (I am not searching for faults, just currious):

``` internal func makeUIView(context: UIViewRepresentableContext<SearchBar>) -> UISearchBar { ``` - one line
``` internal func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) { ``` - multiple lines

🤔

Comment (kaphacius) [reply Demo/Common/Configuration/SearchBar.swift:51]: yeah good point. that one probably should have been one line too :)

**Pull request merged**